tools/java/org/unicode/cldr/tool/WritePluralRules.java

```diff
@@ -13,15 +15,26 @@
 import org.unicode.cldr.util.SupplementalDataInfo.PluralType;
 
 import com.google.common.base.Joiner;
-import com.ibm.icu.dev.util.CollectionUtilities;
+import com.google.common.collect.Comparators;
 import com.ibm.icu.impl.Relation;
 import com.ibm.icu.text.PluralRules;
 
 public class WritePluralRules {
+    // Ordering by size-of-set first, and then lexicographically, with a final tie-break on the
+    // string representation.
+    // Note: This code was refactored from a legacy utility method (which allowed null) but it's
+    // not actually clear if nulls need to be allowed in this case.
+    private static final Comparator<PluralRules> PLURAL_RULES_COMPARATOR =
+        Comparator.<PluralRules, Integer>comparing(r -> r.getKeywords().size())
+            .thenComparing(PluralRules::getKeywords,
+                Comparators.lexicographical(Comparator.<String>nullsFirst(naturalOrder())))
```

Comment:
This and the other "nullsFirst()" call can be removed without breaking anything, and I'd prefer to do that, but you might know why that's not a good idea.